cluster/v1alpha1/types.go

```diff
@@ -233,6 +233,13 @@ type PlacementSpec struct {
 	// Predicates represent a slice of predicates to select ManagedClusters. The predicates are ORed.
 	// +optional
 	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+
+	// PrioritizerConfigs defines the configurations of the prioritizers.
+	// Default configuration is used if it is not specified, in which Steady and Balance prioritizers have the weight of 1
+	// while other prioritizers have the weight of 0.
+	// Referring to PrioritizerConfig to see more description about prioritizer valid names and weight.
+	// +optional
+	PrioritizerConfigs []PrioritizerConfig `json:"prioritizerConfigs,omitempty"`
```

Comment:
@sttts  Can we distinguish between nil and empty on this?

Reply:
`nullable` is a think in openapi. But it is an anti-pattern to distinguish between unset and null. Why is it needed? I would prefer seeing an API that doesn't depend on that. Note that things like JSON merge patch breaks with null.

Reply:
> But it is an anti-pattern to distinguish between unset and null

Between `nil` and `{}`.  Where one could be "no opinion" and the other could be "I have an opinion and want no prioritizers" as opposed to enumerating an ever increasing list..

Reply:
so you actually want `undefined`, `null`, `{}`, all meaning something different?

Reply:
I don't think we need to have undefined different than null.  But I think I see benefit to `null != {}`.  The CRD schema generation allows the distinction?

How does the json merge failure manifest?

Reply:
As discussed with @qiujian16 and @elgnay today, we would like to distinguish between nil and empty. The behavior is:
1. If PrioritizerConfigs is nil (unset), use the default config "steady weight 1, balance weight 1, others weight 0"
2. If PrioritizerConfigs is empty (PrioritizerConfigs: []), disable all the Prioritizers with weight 0.
3. If PrioritizerConfigs is defined, placement will use what user defined (replace default config). That means,
  * If the user only wants to use resourceratioCPU, should config with 
```
  prioritizerConfigs:
    - name: ResourceRatioCPU
      weight: 2
```
  * If the user wants to use resourceratioCPU, as well as the default "steady and balance", should config with 
```
  prioritizerConfigs:
    - name: ResourceRatioCPU
      weight: 1
    - name: Steady
    - name: Balance
```

Reply:
> * If the user wants to use resourceratioCPU, as well as the default "steady and balance", should config with
> 
> ```
>   prioritizerConfigs:
>     - name: ResourceRatioCPU
>       weight: 1
>     - name: Steady
>     - name: Balance
> ```

In this example, why isn't the steady/balance weight set to 1?

Reply:
Otherwise I think this behavior makes sense.

Reply:
As the default weight is 1. The above example has a typo that may make you confused. Below 2 should be the same:
```
  prioritizerConfigs:
    - name: ResourceRatioCPU
      weight: 2
    - name: Steady
    - name: Balance
 ```
```
  prioritizerConfigs:
    - name: ResourceRatioCPU
      weight: 2
    - name: Steady
      weight: 1
    - name: Balance
      weight: 1
 ```